test/Constraints/generics.swift

```diff
@@ -1017,7 +1017,6 @@ func test_requirement_failures_in_ambiguous_context() {
   func f1<T: Equatable>(_: T, _: T) {} // expected-note {{where 'T' = 'A'}}
 
   f1(A(), B()) // expected-error {{local function 'f1' requires that 'A' conform to 'Equatable'}}
-  // expected-error@-1 {{cannot convert value of type 'B' to expected argument type 'A'}}
```

Comment:
This is interesting change, I’d expect either a mismatch or “conflicting arguments to ‘T’” diagnostic here, could you check whether the solver records a fix for that still but doesn’t diagnose it?

Reply:
I’ll take a look. A see a few validation tests still fail so this PR isn’t quite ready yet.